### DISCUSSION TITLE
How to setup Dreambooth to work with 2070Super 8GB VRAM 64GB system RAM Windows11 22H2?

### DISCUSSION THREAD

**Discussion by runner22k** [d8ahazard/sd_dreambooth_extension#785] [Q&A]
**Because of all recent updates, many instructions and tutorials online are obsolete right now, nothing worked for me. I am looking for updated instructions.** There was an old branch/commit, 22Dec, of this extension which used to work for some, even before I could download and install it, it got deleted. So, I am trying to setup dreambooth to work with 2070Super 8GB VRAM 64GB system RAM Windows11 22H2. Does anyone know how to install and get it working? I am trying to install "Kohya_ss" and waiting for someone to help me with the error _"ValueError: fp16 mixed precision requires a GPU."_ 

Meanwhile, is anyone here succeeded to run Dreambooth (not LORA) with 8GB VRAM using this extension?

### UPDATE:
**Maybe it is not possible on windows without deepspeed. With deepspeed I succeeded on linux (Ubuntu 20.04).**

**Comment by runner22k** [comment]
Dear developer @d8ahazard I found deepspeed installation steps and it is installing too. Can you please test and see if it can help to run dreambooth with 8GB cards.

_Originally posted by @PleezDeez in https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/6776#discussioncomment-4700024_


> Deepspeed still isn't working properly on windows and can't be pip installed. I managed to build the whl from source for it tho and it installs but I've reached a new problem. Anytime I try to turn on any sort of distributed training on windows from the accelerate config menu, I get flags from pytorch looking for a linux only backend called NCCL. Here's the build whl I ended up with for it tho https://transfer.sh/eDLOMJ/deepspeed-0.8.0+cd271a4a-cp310-cp310-win_amd64.whl
> But it also requires this one to be installed first https://transfer.sh/me0xpC/triton-2.0.0-cp310-cp310-win_amd64.whl 
> 
> Take both files, drop them in the kohya main folder, right click empty space and open the folder in terminal. Activate the venv with .\venv\Scripts\activate and enter pip install triton-2.0.0-cp310-cp310-win_amd64.whl
> Minimize the terminal, go to your kohya folder then to venv/lib/site-packages and copy the triton folder it creates into the xformers folder also in site-packages and replace any files it conflicts. After that go back to the terminal and pip install deepspeed-0.8.0+cd271a4a-cp310-cp310-win_amd64.whl
> 
> The problem I've ran into now is I need accelerate to actually support the gift I've given them bc if you search the internet for "windows and deepspeed" you'll come back and ask how the hell I managed to do it... I don't know... but I did and it installs with sparse attention support from triton... At least in theory... No way to currently test it til I figure out how to trick accelerate to look for a windows backend instead.

**Reply by cogfilms-3D** [reply]
That Dec 22nd comit was not deleted here is the direct link: - [https://github.com/d8ahazard/sd_dreambooth_extension/tree/c5cb58328c555ac27679422b1da940a9b19de6f2
](https://github.com/d8ahazard/sd_dreambooth_extension/tree/c5cb58328c555ac27679422b1da940a9b19de6f2)

Cogfilms3D

**Reply by runner22k** [reply]
Thanks @cogfilms-3D How to install that check point?

**Reply by cogfilms-3D** [reply]
Its actually very simple. Just download the zip of the commit  and unzip it to the same dreambooth extension in automatic1111. You might want to take a backup of your old dreambooth extension before doing that just in case.

**Reply by cogfilms-3D** [reply]
Well unfortunately the dreambooth extension is broken again, even if you are working with the previous commit trick I used. I suspect its got to do with an update to Automatic1111 that for some reason altered my deambooth extension without being asked to do so. 
I have a feeling that there is no interest in making these tools available for those on 8Gb cards? It would be intersting to know the breakdown of Graphics cards and VRAM that users have to see how much this approach might stiffle development. 

Cogfilms3D

**Reply by runner22k** [reply]
> Its actually very simple. Just download the zip of the commit and unzip it to the same dreambooth extension in automatic1111. You might want to take a backup of your old dreambooth extension before doing that just in case.

Thanks :) and yes dreambooth crashed my Automatic1111 after recent update from Automatic1111. I had to change the gradio version in the requirements folder of dreambooth. 

Which problem did you face with old version of dreambooth because of new updates in Automatic1111?

**Reply by runner22k** [reply]
@cogfilms-3D did you try Kohya_ss? I was able to install Kohya_ss under windows. but I don't know how to optimize it to work with 8GB VRAM.

**Reply by 78Alpha** [reply]
> @cogfilms-3D did you try Kohya_ss? I was able to install Kohya_ss under windows. but I don't know how to optimize it to work with 8GB VRAM. 

https://github.com/ShivamShrirao/diffusers/tree/main/examples/dreambooth#training-on-a-8-gb-gpu

Only a fraction of people have gotten it to work. May need an igpu to get that extra 100 MB VRAM from the main card.

**Reply by runner22k** [reply]
> > @cogfilms-3D did you try Kohya_ss? I was able to install Kohya_ss under windows. but I don't know how to optimize it to work with 8GB VRAM.
> 
> https://github.com/ShivamShrirao/diffusers/tree/main/examples/dreambooth#training-on-a-8-gb-gpu
> 
> Only a fraction of people have gotten it to work. May need an igpu to get that extra 100 MB VRAM from the main card.

**I was able to install ShivamShrirao's version on Ubuntu and was able to train too. But it was taking 2.5 hours for 20 images with 500 regularization images.**

But I wish I can find this **"Changing the default Adam optimizer to DeepSpeed's special version of Adam deepspeed.ops.adam.DeepSpeedCPUAdam gives a substantial speedup but enabling it requires CUDA toolchain with the same version as pytorch. 8-bit optimizer does not seem to be compatible with DeepSpeed at the moment."** to speed up.

**Reply by 78Alpha** [reply]
Everyone's mentions that but you have to modify the optimizer in the script yourself and make other changes that I'm unsure of.

Performance wise. I have a 3070 that can hit 5s/it. Significantly slower than the 1-3 it/s a p40 pulls and further still from a 3070's 16 it/s.
